rtengine/image16.cc

```diff
@@ -284,6 +284,46 @@ void Image16::getStdImage (ColorTemp ctemp, int tran, Imagefloat* image, Preview
 #undef GCLIP
 }
 
+bool
+Image16::isBW() const
+{
+    if (!height || !width) {
+        return false;
+    }
+
+    bool res = true;
+
+    for ( int w = 0, h = height / 2; res && w < width; ++w ) {
+        res =
+            res
+            && (
+                r(h, w) == g(h, w)
+                || r(h, w) == b(h, w)
```

Comment:
@adamreichold I completely agree with you that an early return could reduce the processing time of this loop to about zero in case the image is not b&w!
for ( int w = 0; w < width && res; ++w ) {
would do the job near optimal for this case.
I didn't measure processing time, but for the case of a real b&w image where the loop has to run for all pixels, avoiding the logical operators (which usually introduce branches) should be preferred. I try to post a faster loop soon